pkg/oteltransformationlanguage/ottl/functions.go

```diff
@@ -191,8 +191,8 @@ func (p *Parser) buildArg(argDef Value, argType reflect.Type, index int, args *[
 // never be called in an invocation. Returns whether the arg is an internal arg.
 func (p *Parser) buildInternalArg(argType reflect.Type, args *[]reflect.Value) bool {
 	switch argType.Name() {
-	case "Logger":
```

Comment:
We should probably add this case back so that functions can explicitly ask for a logger and not the entire telemetry settings struct.